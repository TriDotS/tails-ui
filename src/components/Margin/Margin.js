import PropTypes from 'prop-types'
import React from 'react'

const Margin = ({
  margin,
  marginLeft,
  marginRight,
  marginBottom,
  marginTop,
  children
}) => (
  <div
    className={`m-${margin} ml-${marginLeft} mr-${marginRight} mb-${marginBottom} mt-${marginTop}`}
  >
    {children}
  </div>
)

Margin.propTypes = {
  margin: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  marginLeft: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  marginRight: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  marginBottom: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  marginTop: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  children: PropTypes.node
}

export default Margin
