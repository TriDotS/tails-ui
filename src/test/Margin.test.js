import Margin from '../components/Margin/Margin'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Margin>Some Margin</Margin>, div)
})

test('margin snapshot', () => {
  const tree = renderer.create(<Margin>Some Margin</Margin>).toJSON()
  expect(tree).toMatchSnapshot()
})
