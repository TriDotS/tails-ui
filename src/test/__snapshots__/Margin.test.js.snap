// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`margin snapshot 1`] = `
<div
  className="m-undefined ml-undefined mr-undefined mb-undefined mt-undefined"
>
  Some Margin
</div>
`;
